go/vt/wrangler/testlib/shard_test.go

```diff
@@ -0,0 +1,77 @@
+package testlib
+
+import (
+	"strings"
+	"testing"
+
+	"golang.org/x/net/context"
+
+	"github.com/youtube/vitess/go/vt/logutil"
+	"github.com/youtube/vitess/go/vt/tabletmanager/tmclient"
+	"github.com/youtube/vitess/go/vt/topo"
+	"github.com/youtube/vitess/go/vt/vttest/fakesqldb"
+	"github.com/youtube/vitess/go/vt/wrangler"
+	"github.com/youtube/vitess/go/vt/zktopo/zktestserver"
+
+	topodatapb "github.com/youtube/vitess/go/vt/proto/topodata"
+)
+
+func TestDeleteShardCleanup(t *testing.T) {
+	ctx := context.Background()
+	db := fakesqldb.Register()
+	ts := zktestserver.New(t, []string{"cell1", "cell2"})
+	wr := wrangler.New(logutil.NewConsoleLogger(), ts, tmclient.NewTabletManagerClient())
+	vp := NewVtctlPipe(t, ts)
+	defer vp.Close()
+
+	// Create a master, a couple good slaves
+	master := NewFakeTablet(t, wr, "cell1", 0, topodatapb.TabletType_MASTER, db)
+	slave := NewFakeTablet(t, wr, "cell1", 1, topodatapb.TabletType_REPLICA, db)
+	remoteSlave := NewFakeTablet(t, wr, "cell2", 2, topodatapb.TabletType_REPLICA, db)
+
+	// Delete the ShardReplication record in cell2
+	if err := ts.DeleteShardReplication(ctx, "cell2", remoteSlave.Tablet.Keyspace, remoteSlave.Tablet.Shard); err != nil {
+		t.Fatalf("DeleteShardReplication failed: %v", err)
+	}
+
+	// Now try to delete the shard without ignore_if_serving flag,
+	// should fail.
+	if err := vp.Run([]string{
+		"DeleteShard",
+		master.Tablet.Keyspace + "/" + master.Tablet.Shard,
+	}); err == nil || !strings.Contains(err.Error(), "is still serving, cannot delete it") {
+		t.Fatalf("DeleteShard() returned wrong error: %v", err)
+	}
+
+	// Now try to delete the shard without recursive flag, should fail.
```

Comment:
This comment is confusing. "Now try ... without recursive flag" suggests that we've already tried with recursive flag, but I don't see that above.